IdentityCore/src/oauth2/token/MSIDAccessToken.m

```diff
@@ -170,6 +170,21 @@ - (MSIDCredentialType)credentialType
     return MSIDAccessTokenType;
 }
 
+#pragma mark - RefreshNeeded
+
+-(BOOL)refreshNeeded
+{
+    if(self.cachedAt && [[NSDate date] compare:self.cachedAt] == NSOrderedAscending)
+    {
+        return YES;
+    }
+    if(self.refreshOn)
+    {
+        return [self.refreshOn compare:[NSDate date]] == NSOrderedAscending;
```

Comment:
Is there any time zone concerning here? Are they both UTC based?

Reply:
Is the current expires in and extended expires in set by server in same time zone ? I followed the existing implementation for refresh in